Add support for CanadaPostPws

fredericboivin opened the pull request:
CanadaPost is the legacy API

CanadaPostPws is the currently supported API

Some services might be missing, I added the ones currently used/supported by ActiveShipping

fredericboivin committed:
Split package setup by country

Split package setup by country so we can easily test the carriers with
US, Canadian or Europeans destination/origins. Also helps distinguish
between domestic/international addresses

forkata commented on spec/support/shared_contexts/package_setup.rb:
```diff
@@ -32,3 +32,73 @@
   let(:package) { order.shipments.first.to_package }
 end
+
+shared_context 'Canada package setup' do
```
Are these going to be used anywhere in the specs? I like the idea, but a bit hesitant to include here unless they are used.

fredericboivin replied:
It is used in the CanadaPostPws spec to get domestic rates

forkata commented:
@dangerdogz Thanks for adding this! That first commit is a bit large but adding support for this is awesome! We can finally look at deprecating the old Canada Post calculators :+1:

fredericboivin commented:
I will split it in smaller commits

fredericboivin committed:
Add CanadaPostPWS configuration

This commits adds the needed settings for CanadaPostPWS.

fredericboivin committed:
Add CanadaPostPWS calculators

This change adds a basic subset of the main CanadaPostPWS calculators
with some test coverage.

Current carrier is CanadaPost is mostly deprecated, this new carrier is
the currently supported version which works with a modern REST API while
offering far more features.
